common/spec/dependabot/config/ignore_condition_spec.rb

```diff
@@ -172,24 +180,85 @@ def expect_ignored(versions)
         context "with ignore_major_versions" do
           let(:update_types) { ["version-update:semver-major"] }
 
+          # TODO: We probably want to ignore all major version, e.g. v2, v3 etc
+          it "ignores expected versions" do
+            expect_allowed(patch_upgrades + minor_upgrades)
+            expect_ignored(major_upgrades)
+            expect_allowed([dependency_version])
+          end
+
           it "returns the expected range" do
-            expect(ignored_versions).to eq([])
+            expect(ignored_versions).to eq([">= 2.a, < 3"])
           end
         end
 
         context "with ignore_minor_versions" do
           let(:update_types) { ["version-update:semver-minor"] }
 
           it "returns the expected range" do
-            expect(ignored_versions).to eq([])
+            expect(ignored_versions).to eq([">= 1.1.a, < 2"])
           end
         end
 
         context "with ignore_patch_versions" do
           let(:update_types) { ["version-update:semver-patch"] }
+          let(:patch_upgrades) { %w(1.0.2 1.0.5 1.0.4-rc0) }
+          let(:minor_upgrades) { %w(1.1 1.2.0 1.3 1.4.0) }
+
+          it "ignores expected versions" do
+            expect_allowed(major_upgrades + minor_upgrades)
+            expect_ignored(patch_upgrades)
+            expect_allowed([dependency_version, "1.0.0"])
+          end
 
           it "returns the expected range" do
-            expect(ignored_versions).to eq([])
+            expect(ignored_versions).to eq(["> 1, < 1.1"])
+          end
+        end
+      end
+
+      context "with a pre-release semver version" do
+        let(:dependency_version) { "1.2.3-alpha.1" }
```

Comment:
Noice: I dig this additional test case, especially because `parts[2] == "3-alpha"`. That's a spicy meatball to verify!